cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ReplicationPolicyManager.java

```diff
@@ -0,0 +1,119 @@
+
+    public interface ReplicationsCompletedListener {
+        void allReplicationsCompleted();
+
+        void replicationCompleted(int id);
+
+        void replicationErrored(int id);
+    }
+
+    private class ReplicationListener {
+
+        Set<Replicator> replicatorsInProgress;
+
+        ReplicationListener() {
+            replicatorsInProgress = Collections.synchronizedSet(new HashSet<Replicator>());
+        }
+
+        void add(Replicator replicator) {
+            replicatorsInProgress.add(replicator);
+        }
+
+        void remove(Replicator replicator) {
+            if (replicatorsInProgress.remove(replicator)) {
+                replicator.getEventBus().unregister(this);
+            }
+        }
+
+        boolean inProgress(Replicator replicator) {
+            return replicatorsInProgress.contains(replicator);
+        }
+
+        @Subscribe
+        public void complete(ReplicationCompleted event) {
+            finishedReplication(event.replicator);
+            if (mReplicationsCompletedListener != null) {
+                mReplicationsCompletedListener.replicationCompleted(event.replicator.getId());
+            }
+        }
+
+        @Subscribe
+        public void error(ReplicationErrored event) {
+            finishedReplication(event.replicator);
+            if (mReplicationsCompletedListener != null) {
+                mReplicationsCompletedListener.replicationErrored(event.replicator.getId());
+            }
+        }
+
+        public void finishedReplication(Replicator replicator) {
+            synchronized (replicatorsInProgress) {
+                remove(replicator);
+                if (replicatorsInProgress.size() == 0 && mReplicationsCompletedListener != null) {
+                    mReplicationsCompletedListener.allReplicationsCompleted();
+                }
+            }
+        }
+    }
+
+    public ReplicationPolicyManager() {
+        replicators = new ArrayList<Replicator>();
+        replicationListener = new ReplicationListener();
+    }
+
+    public abstract void start();
+
+    public abstract void stop();
+
+    protected void startReplications() {
+        synchronized (replicators) {
+            for (Replicator replicator : replicators) {
+                if (!replicationListener.inProgress(replicator)) {
+                    replicationListener.add(replicator);
+                    replicator.getEventBus().register(replicationListener);
+
+                    replicator.start();
+                }
+            }
+        }
+    }
+
+    protected void stopReplications() {
+        synchronized (replicators) {
+            for (Replicator replicator : replicators) {
+                replicationListener.remove(replicator);
+                replicator.stop();
+            }
+        }
+    }
+
+    public void addReplicators(Replicator... replicators) {
+        synchronized (this.replicators) {
+            if (this.replicators == null) {
```

Comment:
Good point, plus it can never be true anyway as if `this.replicators == null` there would be a NPE on the previous line.  Fixed in 72fbc91.  My intention was that you'd create a new `ReplicationPolicyManager` if you want to change the `Replicator`s associated with a policy, so I don't see the list growing as an issue. I could easily add ability to clear all and/or remove individual replicators if you have a preference for that.